src/e2e-test/java/io/cdap/plugin/cloudsqlpostgresql/stepsdesign/CloudSqlPostGreSql.java

```diff
@@ -188,18 +210,301 @@ public void closeTheCloudSQLPostGreSQLProperties() {
+    }
+
+    @Then("Enter the cloudSQLPostgreSQL properties for database {string} using query {string} " +
+      "for duplicate values {string}")
+    public void enterTheCloudSQLPostgreSQLPropertiesForDatabaseUsingQueryForDuplicateValues
+      (String database, String importQuery, String splitColumnField) throws IOException, InterruptedException {
+        enterTheCloudSQLPostgreSQLPropertiesForDatabase(database);
+        CdfCloudSqlPostGreSqlActions.enterImportQuery(CdapUtils.pluginProp(importQuery));
+        CdfCloudSqlPostGreSqlActions.enterSplitColumn(CdapUtils.pluginProp(splitColumnField));
+    }
+
+    @Then("Enter the cloudSQLPostgreSQL properties for database {string} using query {string} for max values {string}")
+    public void enterTheCloudSQLPostgreSQLPropertiesForDatabaseUsingQueryForMaxValues
+      (String database, String importQuery, String splitColumnField) throws IOException, InterruptedException {
+        enterTheCloudSQLPostgreSQLPropertiesForDatabase(database);
+        CdfCloudSqlPostGreSqlActions.enterImportQuery(CdapUtils.pluginProp(importQuery));
+        CdfCloudSqlPostGreSqlActions.enterSplitColumn(CdapUtils.pluginProp(splitColumnField));
+    }
+
+    @Then("Enter the cloudSQLPostgreSQL properties for database {string} using query {string} for min values {string}")
+    public void enterTheCloudSQLPostgreSQLPropertiesForDatabaseUsingQueryForMinValues
+      (String database, String importQuery, String splitColumnField) throws IOException, InterruptedException {
+      enterTheCloudSQLPostgreSQLPropertiesForDatabase(database);
+     CdfCloudSqlPostGreSqlActions.enterImportQuery(CdapUtils.pluginProp(importQuery));
+     CdfCloudSqlPostGreSqlActions.enterSplitColumn(CdapUtils.pluginProp(splitColumnField));
+    }
+
+    @Then("Enter the cloudSQLPostgreSQL properties for database {string} " +
+      "using query {string} for distinct values {string}")
+    public void enterTheCloudSQLPostgreSQLPropertiesForDatabaseUsingQueryForDistinctValues
+      (String database, String importQuery, String splitColumnField) throws IOException, InterruptedException {
+        enterTheCloudSQLPostgreSQLPropertiesForDatabase(database);
+        CdfCloudSqlPostGreSqlActions.enterImportQuery(CdapUtils.pluginProp(importQuery));
+        CdfCloudSqlPostGreSqlActions.enterSplitColumn(CdapUtils.pluginProp(splitColumnField));
+    }
+
+    @Then("Enter the cloudSQLPostgreSQL properties for database {string} using different join queries {string}")
+    public void enterTheCloudSQLPostgreSQLPropertiesForDatabaseUsingDifferentJoinQueries
+      (String database, String importQuery) throws IOException, InterruptedException {
+        enterTheCloudSQLPostgreSQLPropertiesForDatabase(database);
+        CdfCloudSqlPostGreSqlActions.enterImportQuery(CdapUtils.pluginProp(importQuery));
+    }
+
+    @When("Sink is GCS")
+    public void sinkIsGCS() {
+        CdfStudioActions.sinkGcs();
+    }
+
+    @Then("Validate the output record count")
```

Comment:
validate OUT record count is equals to IN record count

Reply:
done